actioncable/app/assets/javascripts/action_cable/subscription.coffee

```diff
@@ -44,11 +44,9 @@
 # The "AppearanceChannel" name is automatically mapped between the client-side subscription creation and the server-side Ruby class name.
 # The AppearanceChannel#appear/away public methods are exposed automatically to client-side invocation through the @perform method.
 class ActionCable.Subscription
-  constructor: (@subscriptions, params = {}, mixin) ->
+  constructor: (@consumer, params = {}, mixin) ->
     @identifier = JSON.stringify(params)
     extend(this, mixin)
-    @subscriptions.add(this)
```

Comment:
Moved it to `Subscriptions#create` (see ~13 lines down in this PR). Before, `Subscriptions#create` constructed `Subscription` instances with a reference to `@subscriptions` and the subscription would then add itself to the collection. I'm not sure why I implemented that way originally. Now, `Subscriptions#create` [makes a `Subscription` and then adds it to its collection](https://github.com/javan/rails/blob/actioncable-cleanup/actioncable/app/assets/javascripts/action_cable/subscriptions.coffee#L16-L17). 

In short: same end result, less confusing circular object passing.